File: DSA/Krish_Mishra_590015053/Day12_Question3.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
void compressString(char* str) {
    int len = strlen(str);
    char compressed[2 * len + 1];
    int index = 0, count = 1;

    for (int i = 1; i <= len; i++) {
        if (str[i] == str[i - 1]) {
            count++;
        } else {
            compressed[index++] = str[i - 1];
            index += sprintf(&compressed[index], "%d", count);
            count = 1;
        }
    }
    compressed[index] = '\0';

    if (strlen(compressed) < len)
        printf("Output: %s\n", compressed);
    else
        printf("Output: %s\n", str);
}
int main() {
    char input[1000];
    printf("Enter the string: ");
    scanf("%s", input);
    compressString(input);
    return 0;
}

File: DSA/Nivedan_590014904/Day_12_Question_3.py
string = input("Enter a string: ")
newstr = ""
i = 0
while i < len(string):
    count = 1
    while i+1<len(string) and string[i] == string[i+1]:
        count+=1
        i+=1
    newstr += string[i] + str(count)
    i += 1
if len(newstr)>len(string):
       print(string)
else:
        print(newstr)
